### PR TITLE
Extend Vars support.

### LINKED ISSUE #88: Extend Vars support.
In order to allow all Fix expression parameters to include "compile-time" [variables](https://github.com/metafacture/metafacture-core/wiki/Metamorph-User-Guide#parameters-to-metamorph-definitions), I'd like to propose the following changes:

- Introduce new (custom) Fix functions (names up for discussion):
  - `put_var("varName", "value")`: Define single variable (with variable substitution).
  - `put_vars("varName": "value", ...)`: Define multiple variables (without variable substitution¹).
- Update existing expression usages:
  - `do`²/`if`/`elsif`/`unless`: Resolve variable substitutions in parameters.

¹ No support for variable substitution in options required yet.
² Currently there's no bind that accepts any parameters.

### PR DESCRIPTION
**Pull request by blackwinter**
Resolves #88.

### CODE REVIEW

**Review by fsteeg — APPROVED** [review]

**Comment by blackwinter** [comment]
That was quick :) Thanks!

File: metafix/src/main/java/org/metafacture/metafix/Metafix.java [hunk 1]
```diff
@@ -247,6 +243,16 @@ public StreamReceiver getStreamReceiver() {
         return vars;
     }
 
+    public void putVar(final String key, final String value) {
+        try {
+            vars.put(key, value);
+        }
+        catch (final UnsupportedOperationException e) {
+            vars = new HashMap<>(vars);
```
Comment (blackwinter) [inline metafix/src/main/java/org/metafacture/metafix/Metafix.java:251]: @fsteeg: WDYT? The alternative would be to keep a separate map and `putAll(newVars)` in `Metafix(Reader, Map<String, String>)`, like Metamorph does (via `AbstractMetamorphDomWalker`).

Comment (blackwinter) [reply metafix/src/main/java/org/metafacture/metafix/Metafix.java]: Implemented above mentioned alternative in 887d6c6.

**Comment by blackwinter** [comment]
Merging despite unresolved conversation. We can still revert/update later.

**Pull request merged**